presto-main/src/main/java/com/facebook/presto/statistics/TrivialStatisticsCalculator.java

```diff
@@ -0,0 +1,181 @@
+import com.facebook.presto.sql.planner.plan.ExchangeNode;
+import com.facebook.presto.sql.planner.plan.FilterNode;
+import com.facebook.presto.sql.planner.plan.JoinNode;
+import com.facebook.presto.sql.planner.plan.PlanNode;
+import com.facebook.presto.sql.planner.plan.PlanVisitor;
+import com.facebook.presto.sql.planner.plan.ProjectNode;
+import com.facebook.presto.sql.planner.plan.TableScanNode;
+import com.facebook.presto.sql.planner.plan.ValuesNode;
+import com.google.common.collect.ImmutableMap;
+
+import javax.annotation.concurrent.ThreadSafe;
+import javax.inject.Inject;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.facebook.presto.statistics.PlanNodeStatistics.EMPTY_STATISTICS;
+
+@ThreadSafe
+public class TrivialStatisticsCalculator
+        implements StatisticsCalculator
+{
+    private static final Double FILTER_COEFFICIENT = 0.5;
+    private static final Double JOIN_MATCHING_COEEFICIENT = 2.0;
+
+    // todo some computation for outputSizeInBytes
+
+    private final MetadataManager metadataManager;
+
+    @Inject
+    public TrivialStatisticsCalculator(MetadataManager metadataManager)
+    {
+        this.metadataManager = metadataManager;
+    }
+
+    @Override
+    public PlanNodeStatistics calculateStatisticsForNode(Session session, PlanNode planNode)
+    {
+        return calculateStatisticsForPlan(session, planNode).get(planNode);
+    }
+
+    @Override
+    public Map<PlanNode, PlanNodeStatistics> calculateStatisticsForPlan(Session session, PlanNode planNode)
+    {
+        Visitor visitor = new Visitor(session);
+        HashMap<PlanNode, PlanNodeStatistics> statisticsMap = new HashMap<>();
+        planNode.accept(visitor, statisticsMap);
+        return ImmutableMap.copyOf(statisticsMap);
+    }
+
+    private class Visitor
+            extends PlanVisitor<Map<PlanNode, PlanNodeStatistics>, Void>
+    {
+        private final Session session;
+
+        public Visitor(Session session)
+        {
+            this.session = session;
+        }
+
+        @Override
+        protected Void visitPlan(PlanNode node, Map<PlanNode, PlanNodeStatistics> context)
+        {
+            visitChildren(node, context);
+            context.put(node, EMPTY_STATISTICS);
+            return null;
+        }
+
+        private void visitChildren(PlanNode node, Map<PlanNode, PlanNodeStatistics> context)
+        {
+            for (PlanNode source : node.getSources()) {
+                source.accept(this, context);
+            }
+        }
+
+        @Override
+        public Void visitFilter(FilterNode node, Map<PlanNode, PlanNodeStatistics> context)
+        {
+            visitChildren(node, context);
+            PlanNodeStatistics sourceStatistics = context.get(node.getSource());
+            PlanNodeStatistics filterStatistics = sourceStatistics
+                    .mapOutputRowsCount(value -> value * FILTER_COEFFICIENT);
```

Comment:
see todo above. 
I will skip this for now until we decide that we actually like this shape of calculator.